pkg/resources/external_table.go

```diff
@@ -0,0 +1,332 @@
+package resources
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/csv"
+	"fmt"
+	"strings"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+)
+
+const (
+	externalTableIDDelimiter = '|'
+)
+
+var externalTableSchema = map[string]*schema.Schema{
+	"name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Specifies the identifier for the external table; must be unique for the database and schema in which the externalTable is created.",
+	},
+	"schema": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "The schema in which to create the external table.",
+	},
+	"database": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "The database in which to create the external table.",
+	},
+	"column": {
+		Type:        schema.TypeList,
+		Required:    true,
+		MinItems:    1,
+		ForceNew:    true,
+		Description: "Definitions of a column to create in the external table. Minimum one required.",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Column name",
+					ForceNew:    true,
+				},
+				"type": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Column type, e.g. VARIANT",
+					ForceNew:    true,
+				},
+				"as": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "String that specifies the expression for the column. When queried, the column returns results derived from this expression.",
+					ForceNew:    true,
+				},
+			},
+		},
+	},
+	"location": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Specifies a comment for the external table.",
+	},
+	"file_format": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Specifies the file format for the external table.",
+	},
+
+	"aws_sns_topic": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		ForceNew:    true,
+		Description: "Specifies the file format for the external table.",
+	},
+	"partition_by": {
+		Type:        schema.TypeList,
+		Optional:    true,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		ForceNew:    true,
+		Description: "Specifies any partition columns to evaluate for the external table.",
+	},
+	"refresh_on_create": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Specifies to retain the access permissions from the original table when an external table is recreated using the CREATE OR REPLACE TABLE variant",
```

Comment:
I think this was copied from _auto_refresh below.

Reply:
fixed